zookeeper-api/src/main/java/org/apache/helix/zookeeper/constant/ZkSystemPropertyKeys.java

```diff
@@ -63,4 +63,18 @@
 
   /** System property key for jute.maxbuffer */
   public static final String JUTE_MAXBUFFER = "jute.maxbuffer";
+
+  /**
+   * Setting this property to {@code true} in system properties will force Helix ZkClient to use
+   * the <b>non-paginated</b> {@code getChildren} API, no matter if zookeeper supports pagination
+   * or not.
+   * <p>
+   * Given both the zookeeper client and server support <b>paginated</b> {@code getChildren} API as
+   * a prerequisite, if set to {@code false}, it will enable Helix ZkClient's {@code getChildren}
+   * API to call zookeeper's <b>paginated</b> {@code getChildren} API.
+   * <p>
+   * The default value is {@code false}.
+   */
+  public static final String ZK_GETCHILDREN_PAGINATION_DISABLED =
```

Comment:
If our code can automatically detect the new API, why do we need this config option?

Reply:
It's offered to completely disable pagination even though pagination is supported in ZK. It is helpful for deployment: if something is wrong in application after pagination deployed, application can quickly disable pagination by using this system config, without changing the zk lib or roll back zk server.

And if an application wants to use LinkedIn zk but doesn't want to use paginated getChildren, this is an option to disable it as well.

Reply:
Added a note in java doc to indicate the config is temporary - can be deprecated after pagination is deployed and stable.